apis/stargate/v1alpha1/stargate_types.go

```diff
@@ -133,16 +134,19 @@ type StargateDatacenterTemplate struct {
 	Racks []StargateRackTemplate `json:"racks,omitempty"`
 }
 
-// Coalesce compares this StargateDatacenterTemplate with the given StargateClusterTemplate and returns the first
-// non-nil StargateDatacenterTemplate it finds.
-// TODO revisit the merging strategy and/or find a better name for this method
-func (in *StargateDatacenterTemplate) Coalesce(clusterTemplate *StargateClusterTemplate) *StargateDatacenterTemplate {
+// Merge merges the given StargateClusterTemplate with this StargateDatacenterTemplate and returns
+// the result.
+func (in *StargateDatacenterTemplate) Merge(clusterTemplate *StargateClusterTemplate) (*StargateDatacenterTemplate, error) {
 	if in == nil && clusterTemplate == nil {
-		return nil
+		return nil, nil
```

Comment:
> RE treatment of nils: what is Mergo's default behaviour here? Do we need to explicitly handle the case where the left or right inputs are nil, or will it do this for us?

Mergo does not accept nils at all, so this has to be handled beforehand.

> What am I missing?

The field `field1` is unexported. Unexported fields are not processed (because they are not addressable via reflection). Just change your field name to `Field1` and you should be good.

Reply:
Right, that is definitely one cause. I capitalised `Field1` and still see the same results though... 

https://go.dev/play/p/ym6KPol2dJR